File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxProfit = 0;

        int l = 0, r = 0;

        while (r < prices.length){
            int profit = prices[r] - prices[l];
            maxProfit = Math.max(maxProfit, profit );

            if (prices[r] < prices[l]){
                l = r;
            }
            r++;
        }

        return maxProfit;
    }
}

File: Data Structures & Algorithms/maximum-product-subarray/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    maxProduct(nums) {
        let res = nums[0];

        for (let i = 0 ; i < nums.length ; i++){
            let curr = 1;

            for (let j = i ; j < nums.length; j++ ){
                curr *= nums[j];

                res = Math.max(curr, res);
            }
        }

        return res;
    }
}
